tensorflow/contrib/data/python/ops/dataset_ops.py

```diff
@@ -887,6 +887,10 @@ def __init__(self, tensors):
     """See `Dataset.from_tensor_slices()` for details."""
     super(TensorSliceDataset, self).__init__()
     with ops.name_scope("tensors"):
+      tensors = ops.convert_n_to_tensor_or_indexed_slices(tensors)
```

Comment:
Sorry for dropping this PR on the floor... https://github.com/tensorflow/tensorflow/commit/f3f53e8b394bdcaddc707f7bde8dcc98a73531e7 modifies the way nested structures are interpreted to add support for dicts and remove support for lists, so as a result a list will be interpreted as Tensor-like argument and converted to a `tf.Tensor` by the existing code. This fixes the broken doc examples, and I think it satisfies your use cases as well. 

Correct me if I'm wrong, but I think that change supersedes the need for this PR. Is that right?